orchagent/dtelorch.cpp

```diff
@@ -506,7 +511,15 @@ void DTelOrch::doDtelTableTask(Consumer &consumer)
                 if (status != SAI_STATUS_SUCCESS)
                 {
                     SWSS_LOG_ERROR("DTEL ERROR: Failed to enable INT endpoint mode");
-                    goto dtel_table_continue;
+                    if (handleSaiSetStatus(SAI_API_DTEL, status))
+                    {
+                        goto dtel_table_continue;
+                    }
+                    else
+                    {
+                        it++;
```

Comment:
>it++; [](start = 24, length = 5)

I think you change the behavior. Old code will always `erase(it)` no matter success or not.

Reply:
Currently, the handling function always exits orchagent to achieve on-par handling with async mode. So this change will not be activated at the moment and the behavior should be the same as async mode. Once we have a better logic for the failure handling functions, we can fully control whether it should retry or not.